### PR TITLE
Prevent root variable from having white spaces

### PR DESCRIPTION
**Pull request by AlexandreSi**
Also fixes (undetected) issue if users use variable with separator in it (`$.$`)

### CODE REVIEW

File: newIDE/app/src/VariablesList/VariablesList.js [hunk 1]
```diff
@@ -1431,10 +1431,16 @@ const VariablesList = (props: Props) => {
...
       let cleanedName = newName.replace(inheritedPrefix, '');
+
       while (cleanedName.includes(separator)) {
-        cleanedName.replace(separator, '');
+        cleanedName = cleanedName.replace(separator, '');
+      }
+
+      if (depth === 0) {
+        // A root variable cannot have a space in it.
+        cleanedName = cleanedName.replace(/\s/g, '');
```
Comment (ClementPasteau) [inline newIDE/app/src/VariablesList/VariablesList.js:1443]: isn't the logic on "depth" done below in `gd.Project.getSafeName()` ?

File: newIDE/app/src/VariablesList/VariablesList.js [hunk 2]
```diff
@@ -1419,7 +1419,7 @@ const VariablesList = (props: Props) => {
     (newName: string, additionalContext: any) => {
       const parsedContext = JSON.parse(additionalContext);
       const nodeId: string = parsedContext.nodeId;
-      const depth: number = parsedContext.number;
+      const depth: number = parsedContext.depth;
```
Comment (4ian) [inline newIDE/app/src/VariablesList/VariablesList.js:1422]: 🤦‍♂️ 
Of course I had to make a mistake in the only non typed part of the editor

File: newIDE/app/src/VariablesList/VariablesList.js (hunk 1 above)
Comment (4ian) [reply newIDE/app/src/VariablesList/VariablesList.js:1443]: Indeed, this should be handled below

Comment (AlexandreSi) [reply newIDE/app/src/VariablesList/VariablesList.js]: Indeed!

File: newIDE/app/src/VariablesList/VariablesList.js (hunk 2 above)
Comment (ClementPasteau) [reply newIDE/app/src/VariablesList/VariablesList.js:1422]: duh

**Pull request merged**